java/ql/lib/semmle/code/java/environment/SystemProperty.qll

```diff
@@ -0,0 +1,249 @@
+  |
+    m.hasName(["getTempDirectory", "getTempDirectoryPath"]) and propertyName = "java.io.tmpdir"
+    or
+    m.hasName(["getUserDirectory", "getUserDirectoryPath"]) and propertyName = "user.home"
+  )
+}
+
+private MethodAccess getSystemPropertyFromGuava(string propertyName) {
+  exists(EnumConstant ec |
+    ec.getDeclaringType().hasQualifiedName("com.google.common.base", "StandardSystemProperty") and
+    result.getQualifier() = ec.getAnAccess() and
+    result.getMethod().hasName("value")
+  |
+    ec.hasName("JAVA_VERSION") and propertyName = "java.version"
+    or
+    ec.hasName("JAVA_VENDOR") and propertyName = "java.vendor"
+    or
+    ec.hasName("JAVA_VENDOR_URL") and propertyName = "java.vendor.url"
+    or
+    ec.hasName("JAVA_HOME") and propertyName = "java.home"
+    or
+    ec.hasName("JAVA_VM_SPECIFICATION_VERSION") and propertyName = "java.vm.specification.version"
+    or
+    ec.hasName("JAVA_VM_SPECIFICATION_VENDOR") and propertyName = "java.vm.specification.vendor"
+    or
+    ec.hasName("JAVA_VM_SPECIFICATION_NAME") and propertyName = "java.vm.specification.name"
+    or
+    ec.hasName("JAVA_VM_VERSION") and propertyName = "java.vm.version"
+    or
+    ec.hasName("JAVA_VM_VENDOR") and propertyName = "java.vm.vendor"
+    or
+    ec.hasName("JAVA_VM_NAME") and propertyName = "java.vm.name"
+    or
+    ec.hasName("JAVA_SPECIFICATION_VERSION") and propertyName = "java.specification.version"
+    or
+    ec.hasName("JAVA_SPECIFICATION_VENDOR") and propertyName = "java.specification.vendor"
+    or
+    ec.hasName("JAVA_SPECIFICATION_NAME") and propertyName = "java.specification.name"
+    or
+    ec.hasName("JAVA_CLASS_VERSION") and propertyName = "java.class.version"
+    or
+    ec.hasName("JAVA_CLASS_PATH") and propertyName = "java.class.path"
+    or
+    ec.hasName("JAVA_LIBRARY_PATH") and propertyName = "java.library.path"
+    or
+    ec.hasName("JAVA_IO_TMPDIR") and propertyName = "java.io.tmpdir"
+    or
+    ec.hasName("JAVA_COMPILER") and propertyName = "java.compiler"
+    or
+    ec.hasName("JAVA_EXT_DIRS") and propertyName = "java.ext.dirs"
+    or
+    ec.hasName("OS_NAME") and propertyName = "os.name"
+    or
+    ec.hasName("OS_ARCH") and propertyName = "os.arch"
+    or
+    ec.hasName("OS_VERSION") and propertyName = "os.version"
+    or
+    ec.hasName("FILE_SEPARATOR") and propertyName = "file.separator"
+    or
+    ec.hasName("PATH_SEPARATOR") and propertyName = "path.separator"
+    or
+    ec.hasName("LINE_SEPARATOR") and propertyName = "line.separator"
+    or
+    ec.hasName("USER_NAME") and propertyName = "user.name"
+    or
+    ec.hasName("USER_HOME") and propertyName = "user.home"
+    or
+    ec.hasName("USER_DIR") and propertyName = "user.dir"
+  )
+}
+
+private MethodAccess getSystemPropertyFromOperatingSystemMXBean(string propertyName) {
+  exists(Method m |
+    m = result.getMethod() and
+    m.getDeclaringType().hasQualifiedName("java.lang.management", "OperatingSystemMXBean")
```

Comment:
Might be good to consider subtypes as well, because there is at least [`com.sun.management.OperatingSystemMXBean`](https://docs.oracle.com/en/java/javase/17/docs/api/jdk.management/com/sun/management/OperatingSystemMXBean.html)

Reply:
Ah you are right; `com.sun.management.OperatingSystemMXBean` does not override them, and there are no other subtypes which override these either. You could of course consider declarations by subtypes nonetheless in case the JDK overrides them at some point (e.g. to adjust Javadoc), but it is probably unlikely that this will happen.